ui/components/app/snaps/snap-ui-avatar/snap-ui-avatar.tsx

```diff
@@ -1,44 +1,16 @@
-import React, { useMemo } from 'react';
-import { useSelector } from 'react-redux';
-import { CaipAccountId, parseCaipAccountId } from '@metamask/utils';
-import BlockieIdenticon from '../../../ui/identicon/blockieIdenticon';
-import Jazzicon from '../../../ui/jazzicon';
-import { getUseBlockie } from '../../../../selectors';
+import React from 'react';
+import { AvatarAccountSize } from '@metamask/design-system-react';
+import { PreferredAvatar } from '../../preferred-avatar';
 
-export const DIAMETERS: Record<string, number> = {
-  xs: 16,
-  sm: 24,
-  md: 32,
-  lg: 40,
-};
-
-export type SnapUIAvatarProps = {
+type SnapUIAvatarProps = {
   // The address must be a CAIP-10 string.
   address: string;
-  size?: 'xs' | 'sm' | 'md' | 'lg';
+  size?: AvatarAccountSize;
```

Comment:
One thing we may want to check is whether we can still use this component via the Snaps SDK / Snaps API now that the type has changed from a string to an enum. Since the Snap UI is accessed through the Snaps API, we may need to continue supporting string types. Let's confirm with the @MetaMask/snaps team

Reply:
It's fine to use the `AvatarAccountSize` enum here :)